plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2722,7 +2725,16 @@ public OptionalLong executeDelete(ConnectorSession session, ConnectorTableHandle
             metastore.truncateUnpartitionedTable(session, handle.getSchemaName(), handle.getTableName());
         }
         else {
-            for (HivePartition hivePartition : partitionManager.getOrLoadPartitions(metastore, handle)) {
+            List<HivePartition> partitionsToBeDropped = partitionManager.getOrLoadPartitions(metastore, handle);
```

Comment:
When we're checking at this point, it doesn't yet count towards the transaction. For example, say if the limit is `2`, and the following sequence would still succeed even though we're deleting 3 partitions:

```
create table xyz ( a bigint, b bigint) with (partitioned_by = array['b']);
insert into xyz values (5, 5), (6, 6), (7, 7);
START TRANSACTION;
DELETE FROM xyz WHERE b = 5;
DELETE FROM xyz WHERE b = 6;
DELETE FROM xyz WHERE b = 7;
COMMIT;
```

this would succeed. So just calling this `max-partition-drops` would be better. Other option would be to have this check inside the `Semitransactionalhivemetastore`, but it'd be simpler to align with the `max-partitions-per-scan`.

Reply:
makes sense, it's more like `max-partition-drops-per-query`